frontend/ui/widget/textboxwidget.lua

```diff
@@ -333,4 +361,121 @@ function TextBoxWidget:onHoldWord(callback, ges)
     return
 end
 
+-- Allow selection of one or more words (with no visual feedback)
+-- Gestures should be declared in widget using us (e.g dictquicklookup.lua)
+function TextBoxWidget:onHoldStartText(_, ges)
+    -- just store hold start position and timestamp, will be used on release
+    self.hold_start_x = ges.pos.x - self.dimen.x
+    self.hold_start_y = ges.pos.y - self.dimen.y
+    self.hold_start_tv = TimeVal.now()
+    return true
+end
+
+function TextBoxWidget:onHoldReleaseText(callback, ges)
+    if not callback then return end
+
+    local hold_end_x = ges.pos.x - self.dimen.x
+    local hold_end_y = ges.pos.y - self.dimen.y
+    local hold_duration = TimeVal.now() - self.hold_start_tv
+    hold_duration = hold_duration.sec + hold_duration.usec/1000000
+
+    -- Swap start and end if needed
+    local x0, y0, x1, y1
+    -- first, sort by y/line_num
+    local start_line_num = math.ceil(self.hold_start_y / self.line_height_px)
+    local end_line_num = math.ceil(hold_end_y / self.line_height_px)
+    if start_line_num < end_line_num then
+        x0, y0 = self.hold_start_x, self.hold_start_y
+        x1, y1 = hold_end_x, hold_end_y
+    elseif start_line_num > end_line_num then
+        x0, y0 = hold_end_x, hold_end_y
+        x1, y1 = self.hold_start_x, self.hold_start_y
+    else -- same line_num : sort by x
+        if self.hold_start_x <= hold_end_x then
+            x0, y0 = self.hold_start_x, self.hold_start_y
+            x1, y1 = hold_end_x, hold_end_y
+        else
+            x0, y0 = hold_end_x, hold_end_y
+            x1, y1 = self.hold_start_x, self.hold_start_y
+        end
+    end
+
+    -- similar code to find start or end is in _findWordEdge() helper
+    local sel_start_idx = self:_findWordEdge(x0, y0, "start")
```

Comment:
Better use module global int flags for things like this, it's much more efficient for comparison:

```lua
local FIND_START = 1
local FIND_END = 2
```